datashare-app/src/main/java/org/icij/datashare/com/mail/MailSender.java

```diff
@@ -82,16 +97,21 @@ private Message createMessage(Mail donneesEmail) throws MessagingException, Addr
         return message;
     }
 
-    private synchronized static Session getMailSession(String hostTransportMail, int portTransportMail, String user, String password) {
+    private synchronized static Session getMailSession(MailSender mailSender) {
         Properties properties = new Properties();
         properties.setProperty("mail.smtp.class", "com.sun.mail.smtp.SMTPTransport");
-        properties.setProperty("mail.smtp.port", String.valueOf(portTransportMail));
-        properties.setProperty("mail.smtp.host", hostTransportMail);
-        return user == null ? Session.getDefaultInstance(properties): Session.getInstance(properties, new Authenticator() {
+        properties.setProperty("mail.smtp.port", String.valueOf(mailSender.port));
+        properties.setProperty("mail.smtp.host", mailSender.host);
+        properties.setProperty("mail.smtp.auth", String.valueOf(mailSender.shouldAuth()));
+        properties.setProperty("mail.smtp.ssl.enable", String.valueOf(mailSender.tls));
```

Comment:
Shouldn't we have some sort of validation to ensure that String value of boolean are either "true" or "false"?

Reply:
except if I don't understand everything, java boolean type is guarantying that the value cannot be something else than true/false ?

Reply:
plus there are UT's on auth.